### PR TITLE
Fix transfer failure return result

### PR DESCRIPTION
**Pull request by ninjaahhh**
in normal case this code path won't be touched because:

1. normal transfer tx will be guarded by validation
2. transfer / calls in smart contract will be guarded by VM checking QKC balance

thus there are two ways to break the execution rule:

1. indirectly make a CALL msg (outside the VM) such as calling transfer_mnt precompiled contract (fixed by this PR)
~2. make a CALL message with value transfer but in non-default token~

### CODE REVIEW

File: fixtures/QuarkChainStateTests/stContractCallMnt/contractTransferMntNotEnoughBalance.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      "ConstantinopleFix": [
+        {
+          "hash": "0xf5811632e06923cfbbd8b8d2fc8dcd13dd0c583f96d385f203d423fd73bee80c",
+          "indexes": {
+            "data": 0,
+            "gas": 0,
+            "value": 0,
+            "transferTokenId": 0
+          },
+          "logs": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347"
```
Comment (ninjaahhh) [inline fixtures/QuarkChainStateTests/stContractCallMnt/contractTransferMntNotEnoughBalance.json]: no need for log

File: fixtures/QuarkChainStateTests/stContractCallMnt/contractTransferMntNotEnoughBalance.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        },
+        {
+          "hash": "0xf5811632e06923cfbbd8b8d2fc8dcd13dd0c583f96d385f203d423fd73bee80c",
```
Comment (ninjaahhh) [inline fixtures/QuarkChainStateTests/stContractCallMnt/contractTransferMntNotEnoughBalance.json]: same hash as first one. can remove?

**Comment by ninjaahhh** [comment]
cc @pingke this is a consensus change for current testnet3, please make sure the new test can pass on your side

File: quarkchain/evm/messages.py [hunk 3]
```diff
@@ -560,8 +560,7 @@ def _apply_msg(ext, msg, code):
                 have=ext.get_balance(msg.sender, token_id=msg.transfer_token_id),
                 want=msg.value,
             )
-            # TODO: Why return success if the transfer failed?
-            return 1, msg.gas, []
+            return 0, 0, []
```
Comment (qizhou) [inline quarkchain/evm/messages.py]: May we double check how goetherum works on this to make sure the behavior is consistent?  In addition, I am wondering if there is an existing test in eth repo to test it?

Comment (ninjaahhh) [reply quarkchain/evm/messages.py]: as mentioned in my first comment, in normal cases this should be dead code. however in our system it becomes possible in transfer_mnt precompiled contract (or if we have a bug on `apply_xshard_deposit`)

**Pull request merged**